iothub/device/devdoc/Convention-based operations.md

```diff
@@ -0,0 +1,213 @@
+}
+
+public sealed class DefaultPayloadConvention : PayloadConvention {
+    public static readonly DefaultPayloadConvention Instance;
+    public DefaultPayloadConvention();
+    public override PayloadEncoder PayloadEncoder { get; }
+    public override PayloadSerializer PayloadSerializer { get; }
+}
+
+public class Utf8PayloadEncoder : PayloadEncoder {
+    public static readonly Utf8PayloadEncoder Instance;
+    public Utf8PayloadEncoder();
+    public override Encoding ContentEncoding { get; }
+    public override byte[] EncodeStringToByteArray(string contentPayload);
+}
+
+public class NewtonsoftJsonPayloadSerializer : PayloadSerializer {
+    public static readonly NewtonsoftJsonPayloadSerializer Instance;
+    public NewtonsoftJsonPayloadSerializer();
+    public override string ContentType { get; }
+    public override T ConvertFromObject<T>(object objectToConvert);
+    public override IWritablePropertyResponse CreateWritablePropertyResponse(object value, int statusCode, long version, string description = null);
+    public override T DeserializeToType<T>(string stringToDeserialize);
+    public override string SerializeToString(object objectToSerialize);
+    public override bool TryGetNestedObjectValue<T>(object objectToConvert, string propertyName, out T outValue);
+}
+
+public abstract class PayloadCollection : IEnumerable, IEnumerable<object> {
+    protected PayloadCollection();
+    public IDictionary<string, object> Collection { get; private set; }
+    public PayloadConvention Convention { get; internal set; }
+    public virtual object this[string key] { get; set; }
+    public virtual void Add(string key, object value);
+    public virtual void AddOrUpdate(string key, object value);
+    public bool Contains(string key);
+    public IEnumerator<object> GetEnumerator();
+    public virtual byte[] GetPayloadObjectBytes();
+    public virtual string GetSerializedString();
+    protected void SetCollection(PayloadCollection payloadCollection);
+    IEnumerator System.Collections.IEnumerable.GetEnumerator();
+    public bool TryGetValue<T>(string key, out T value);
+}
+
+public static class ConventionBasedConstants {
+    public const string AckCodePropertyName = "ac";
+    public const string AckDescriptionPropertyName = "ad";
+    public const string AckVersionPropertyName = "av";
+    public const string ComponentIdentifierKey = "__t";
+    public const string ComponentIdentifierValue = "c";
+    public const string ValuePropertyName = "value";
+}
+```
+
+### Properties
+
+```csharp
+/// <summary>
+/// Retrieve the device properties.
+/// </summary>
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+/// <returns>The device properties.</returns>
+public Task<ClientProperties> GetClientPropertiesAsync(CancellationToken cancellationToken = default);
+
+/// <summary>
+/// Update properties.
```

Comment:
Update device properties?

Reply:
this could be for either device or module clients though, so we're not specifying anything here